src/plugins/data/server/index_patterns/index_patterns_api_client.ts

```diff
@@ -27,12 +28,23 @@ export class IndexPatternsApiServer implements IIndexPatternsApiClient {
     allowNoIndex,
   }: GetFieldsOptions) {
     const indexPatterns = new IndexPatternsFetcher(this.esClient, allowNoIndex);
-    return await indexPatterns.getFieldsForWildcard({
-      pattern,
-      metaFields,
-      type,
-      rollupIndex,
-    });
+    return await indexPatterns
+      .getFieldsForWildcard({
+        pattern,
+        metaFields,
+        type,
+        rollupIndex,
+      })
+      .catch((err) => {
+        if (
+          err.output.payload.statusCode === 404 &&
+          err.output.payload.code === 'no_matching_indices'
+        ) {
+          throw new IndexPatternMissingIndices(pattern);
+        } else {
+          throw err;
```

Comment:
When this is thrown, does the error get caught and logged to the server logs somewhere? 

Or should there be a `logger.error(err)` before the throw?

Reply:
It gets caught and logged to the server logs.